src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,227 @@
+                InvalidUserDataException.class,
+                () -> registrationService.register(user)
+        );
+        assertEquals(SHORT_PASSWORD_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    public void register_underageUser_notOk() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, UNDERAGE);
+
+        InvalidUserDataException exception = assertThrows(
+                InvalidUserDataException.class,
+                () -> registrationService.register(user)
+        );
+        assertEquals(UNDERAGE_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    public void register_nullLogin_notOk() {
+        User user = createUser(null, VALID_PASSWORD, VALID_AGE);
+
+        InvalidUserDataException exception = assertThrows(
+                InvalidUserDataException.class,
+                () -> registrationService.register(user)
+        );
+        assertEquals(NULL_LOGIN_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    public void register_nullPassword_notOk() {
+        User user = createUser(VALID_LOGIN, null, VALID_AGE);
+
+        InvalidUserDataException exception = assertThrows(
+                InvalidUserDataException.class,
+                () -> registrationService.register(user)
+        );
+        assertEquals(NULL_PASSWORD_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    public void register_nullAge_notOk() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, null);
+
+        InvalidUserDataException exception = assertThrows(
+                InvalidUserDataException.class,
+                () -> registrationService.register(user)
+        );
+        assertEquals(NULL_AGE_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    public void register_emptyLogin_notOk() {
+        User user = createUser("", VALID_PASSWORD, VALID_AGE);
+
+        InvalidUserDataException exception = assertThrows(
+                InvalidUserDataException.class,
+                () -> registrationService.register(user)
+        );
+        assertEquals(SHORT_LOGIN_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    public void register_emptyPassword_notOk() {
+        User user = createUser(VALID_LOGIN, "", VALID_AGE);
+
+        InvalidUserDataException exception = assertThrows(
+                InvalidUserDataException.class,
+                () -> registrationService.register(user)
+        );
+        assertEquals(SHORT_PASSWORD_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    public void register_negativeAge_notOk() {
+        User user = createUser(VALID_LOGIN, VALID_PASSWORD, NEGATIVE_AGE);
+
+        InvalidUserDataException exception = assertThrows(
+                InvalidUserDataException.class,
+                () -> registrationService.register(user)
+        );
+        assertEquals(UNDERAGE_MESSAGE, exception.getMessage());
+    }
+
+    @Test
+    public void register_edgeCasePasswords_okAndNotOk() {
```

Comment:
You shouldn't connect `ok` tests with `notOk`. Split it to 2 separate test methods.